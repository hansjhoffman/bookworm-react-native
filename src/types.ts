export type Maybe<T> = T | null;
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string,
  String: string,
  Boolean: boolean,
  Int: number,
  Float: number,
};

export type Book = {
   __typename?: 'Book',
  /** Combined authors of the book */
  authors: Scalars['String'],
  /** A short summary of the book */
  description?: Maybe<Scalars['String']>,
  /** Simple auto-increment database id */
  id: Scalars['ID'],
  /** A random rating between 0 and 5 */
  rating: Scalars['Int'],
  /** The uri for a thumbnail */
  thumbnail: Scalars['String'],
  /** The title of the book */
  title: Scalars['String'],
};

export type RootQueryType = {
   __typename?: 'RootQueryType',
  /** Get info about a book */
  book?: Maybe<Book>,
  /** List all books */
  books?: Maybe<Array<Maybe<Book>>>,
};


export type RootQueryTypeBookArgs = {
  id: Scalars['ID']
};
