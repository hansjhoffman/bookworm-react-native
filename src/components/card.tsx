import React from "react";
import {
  Image,
  Platform,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from "react-native";

import { Stars } from "./stars";
import { Colors, Layout } from "../constants";
import { Book } from "../types";
import { Images } from "../../assets";

type Props = {
  book: Book;
  onPress?: () => void;
};

export function Card({ book, onPress }: Props) {
  return (
    <TouchableOpacity style={styles.card} onPress={onPress}>
      <View style={styles.thumbnailContainer}>
        <Image
          defaultSource={Images.placeholder}
          resizeMode="contain"
          source={{ uri: book.thumbnail ? book.thumbnail : Images.placeholder }}
          style={styles.thumbnail}
        />
      </View>
      <View style={styles.infoContainer}>
        <Text style={styles.bookTitleText} numberOfLines={1}>
          {book.title}
        </Text>
        <Text style={styles.bookAuthorText} numberOfLines={1}>
          {book.authors}
        </Text>
      </View>
      <Stars rating={book.rating} />
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  bookAuthorText: {
    color: Colors.textLight,
    fontFamily: "Lato-Light",
    fontSize: 14,
    marginBottom: Layout.margin_sm,
  },
  bookTitleText: {
    color: Colors.textDark,
    fontFamily: "Lato-Bold",
    fontSize: 18,
    marginBottom: Layout.margin_sm,
  },
  card: {
    backgroundColor: Colors.white,
    borderRadius: Layout.cardRadius,
    height: 200,
    marginBottom: 100,
    padding: Layout.margin_md,
    position: "relative",
    width: (Layout.screen.width - Layout.margin_md * 3) / 2,
    ...Platform.select({
      android: {
        elevation: 5,
      },
      ios: {
        shadowColor: Colors.shadow,
        shadowOffset: {
          height: 7,
          width: 0,
        },
        shadowOpacity: 0.1,
        shadowRadius: Layout.cardRadius,
      },
    }),
  },
  infoContainer: {
    paddingTop: 90,
  },
  thumbnail: {
    borderRadius: 15,
    height: 200,
    overflow: "hidden",
    position: "absolute",
    top: -100,
    width: 100,
  },
  thumbnailContainer: {
    alignItems: "center",
  },
});
