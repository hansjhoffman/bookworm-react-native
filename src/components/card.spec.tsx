import React from "react";
import { render } from "@testing-library/react-native";

import { Card } from "./card";
import { Book } from "../types";

describe("[component] Card", function() {
  const book: Book = {
    id: "1",
    authors: "Ayn Rand",
    description: "This book is one of my top 5!",
    rating: 3,
    thumbnail: "https://thumbnail.jpg",
    title: "Atlas Shrugged",
  };

  it("should match snapshot", function() {
    const { baseElement } = render(<Card book={book} />);

    expect(baseElement).toMatchSnapshot();
  });
});
