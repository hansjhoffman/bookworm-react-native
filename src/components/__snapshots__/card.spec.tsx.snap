// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[component] Card should match snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 8,
          "elevation": 1,
          "height": 200,
          "marginBottom": 100,
          "padding": 15,
          "position": "relative",
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 7,
            "width": 7,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 8,
          "width": 352.5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <Image
          defaultSource={
            Object {
              "testUri": "../../../assets/images/placeholder.png",
            }
          }
          resizeMode="contain"
          source={
            Object {
              "uri": "https://thumbnail.jpg",
            }
          }
          style={
            Object {
              "borderRadius": 15,
              "height": 200,
              "position": "absolute",
              "top": -100,
              "width": 100,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "paddingTop": 90,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "#333333",
              "fontFamily": "Lato-Bold",
              "fontSize": 18,
              "marginBottom": 10,
            }
          }
        >
          Atlas Shrugged
        </Text>
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "#b4b4b4",
              "fontFamily": "Lato-Light",
              "fontSize": 14,
              "marginBottom": 10,
            }
          }
        >
          Ayn Rand
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../assets/icons/star.png",
            }
          }
          style={
            Object {
              "height": 18,
              "marginLeft": 0,
              "tintColor": "#f4ca67",
              "width": 18,
            }
          }
        />
        <Image
          source={
            Object {
              "testUri": "../../../assets/icons/star.png",
            }
          }
          style={
            Object {
              "height": 18,
              "marginLeft": 1.5,
              "tintColor": "#f4ca67",
              "width": 18,
            }
          }
        />
        <Image
          source={
            Object {
              "testUri": "../../../assets/icons/star.png",
            }
          }
          style={
            Object {
              "height": 18,
              "marginLeft": 1.5,
              "tintColor": "#f4ca67",
              "width": 18,
            }
          }
        />
        <Image
          source={
            Object {
              "testUri": "../../../assets/icons/star.png",
            }
          }
          style={
            Object {
              "height": 18,
              "marginLeft": 1.5,
              "tintColor": "#dbdbdb",
              "width": 18,
            }
          }
        />
        <Image
          source={
            Object {
              "testUri": "../../../assets/icons/star.png",
            }
          }
          style={
            Object {
              "height": 18,
              "marginLeft": 1.5,
              "tintColor": "#dbdbdb",
              "width": 18,
            }
          }
        />
      </View>
    </TouchableOpacity>
  </View>
</View>
`;
